src/gpuarray_array_collectives.c

```diff
@@ -0,0 +1,108 @@
+#include "gpuarray/array.h"
+#include "gpuarray/buffer_collectives.h"
+#include "gpuarray/collectives.h"
+#include "gpuarray/error.h"
+
+#include "private.h"
+
+static inline int find_total_elems(const GpuArray* array)
+{
```

Comment:
Eventually nccl functions will get an `int count`. Should I convert later? Should we put a constraint for not too large arrays?

https://github.com/NVIDIA/nccl/blob/master/src/nccl.h#L130